static/app/views/performance/newTraceDetails/traceApi/useTraceRootEvent.tsx

```diff
@@ -1,26 +1,60 @@
 import type {EventTransaction} from 'sentry/types/event';
-import {isTraceSplitResult} from 'sentry/utils/performance/quickTrace/utils';
-import {useApiQuery} from 'sentry/utils/queryClient';
+import {useApiQuery, type UseApiQueryResult} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
 import useOrganization from 'sentry/utils/useOrganization';
 import {useSyncedLocalStorageState} from 'sentry/utils/useSyncedLocalStorageState';
+import {
+  type TraceItemDetailsResponse,
+  useTraceItemDetails,
+} from 'sentry/views/explore/hooks/useTraceItemDetails';
+import {
+  OurLogKnownFieldKey,
+  type OurLogsResponseItem,
+} from 'sentry/views/explore/logs/types';
+import {TraceItemDataset} from 'sentry/views/explore/types';
+import {getRepresentativeTraceEvent} from 'sentry/views/performance/newTraceDetails/traceApi/utils';
 import {TRACE_FORMAT_PREFERENCE_KEY} from 'sentry/views/performance/newTraceDetails/traceHeader/styles';
 import type {TraceTree} from 'sentry/views/performance/newTraceDetails/traceModels/traceTree';
 
-export function useTraceRootEvent(trace: TraceTree.Trace | null) {
-  const root = trace
-    ? isTraceSplitResult(trace)
-      ? trace?.transactions?.[0] || trace?.orphan_errors?.[0]
-      : trace[0]
-    : null;
+type Params = {
+  logs: OurLogsResponseItem[] | undefined;
+  traceId: string;
+  tree: TraceTree;
+};
+
+export type TraceRootEventQueryResults =
+  | UseApiQueryResult<EventTransaction, RequestError>
+  | UseApiQueryResult<TraceItemDetailsResponse, RequestError>;
+
+export function useTraceRootEvent({
+  tree,
+  logs,
+  traceId,
+}: Params): TraceRootEventQueryResults {
+  const rep = getRepresentativeTraceEvent(tree, logs);
   const organization = useOrganization();
   const [storedTraceFormat] = useSyncedLocalStorageState(
     TRACE_FORMAT_PREFERENCE_KEY,
     'non-eap'
   );
 
-  return useApiQuery<EventTransaction>(
+  // TODO: This is a bit of a mess, we won't need all of this once we switch to EAP only
+  const treeIsLoading = tree.type === 'loading';
+  const hasOnlyLogs = !!(tree.type === 'empty' && logs && logs.length > 0);
+  const enabledBase =
+    !treeIsLoading &&
+    (tree.type === 'trace' || hasOnlyLogs) &&
+    !!rep &&
+    !!rep.event &&
```

Comment:
typescript says `rep` is never undefined, so you can probably skip checking for `!!rep`. `!!rep?event` should also work if you prefer that

Reply:
Valid point, I'll push the change